pkg/exporters/stackdriver/gce/type.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"cloud.google.com/go/compute/metadata"
+	"github.com/golang/glog"
+)
+
+type Metadata struct {
+	ProjectID    string `json:"projectID"`
+	Zone         string `json:"zone"`
+	InstanceID   string `json:"instanceID"`
+	InstanceName string `json:"instanceName"`
+}
+
+func (md *Metadata) IsValid() bool {
```

Comment:
Better return error to indicate which metadata is missing.

Reply:
I don't really think returning the list of missing fields would be useful.

This is because if any field is missing, we will just call `PopulateFromGCE` and fill it. If `PopulateFromGCE` failed, it will generate an error message then.

I changed the function to `HasMissingField()`, maybe that can help make the logic a bit clearer?